cfme/provisioning.py

```diff
@@ -35,6 +40,126 @@ def select_security_group(sg):
+        provision_type = BootstrapSelect('service__provision_type')
+        linked_clone = Input(name='service__linked_clone')
+        pxe_server = BootstrapSelect('service__pxe_server_id')
+        pxe_image = Table('//div[@id="prov_pxe_img_div"]/table')
+        iso_file = Table('//div[@id="prov_iso_img_div"]/table')
+
+    @View.nested
+    class environment(Tab):  # noqa
+        TAB_NAME = 'Environment'
+
+        automatic_placement = Input(name='environment__placement_auto')
+        # Cloud
+        availability_zone = BootstrapSelect('environment__placement_availability_zone')
+        cloud_network = BootstrapSelect('environment__cloud_network')
+        cloud_subnet = BootstrapSelect('environment__cloud_subnet')
+        security_groups = BootstrapSelect('environment__security_groups')
+        resource_groups = BootstrapSelect('environment__resource_group')
+        public_ip_address = BootstrapSelect('environment__floating_ip_address')
+        # Infra
+        provider_name = BootstrapSelect('environment__placement_ems_name')
+        datacenter = BootstrapSelect('environment__placement_dc_name')
+        cluster = BootstrapSelect('environment__placement_cluster_name')
+        resource_pool = BootstrapSelect('environment__placement_rp_name')
+        folder = BootstrapSelect('environment__placement_folder_name')
+        host_filter = BootstrapSelect('environment__host_filter')
+        host_name = Table('//div[@id="prov_host_div"]/table')
+        datastore_create = Input('environment__new_datastore_create')
+        datastore_filter = BootstrapSelect('environment__ds_filter')
+        datastore_name = Table('//div[@id="prov_ds_div"]/table')
+
+    @View.nested
+    class hardware(Tab):  # noqa
+        TAB_NAME = 'Hardware'
+        num_sockets = BootstrapSelect('hardware__number_of_sockets')
+        cores_per_socket = BootstrapSelect('hardware__cores_per_socket')
+        num_cpus = BootstrapSelect('hardware__number_of_cpus')
+        memory = BootstrapSelect('hardware__vm_memory')
+        # TODO radio widget # disk_format', ui.Radio('hardware__disk_format')
+        vm_limit_cpu = Input(name='hardware__cpu_limit')
+        vm_limit_memory = Input(name='hardware__memory_limit')
+        vm_reserve_cpu = Input(name='hardware__cpu_reserve')
+        vm_reserve_memory = Input(name='hardware__memory_reserve')
+
+    @View.nested
+    class network(Tab):  # noqa
+        TAB_NAME = 'Network'
+        vlan = BootstrapSelect('network__vlan')
+
+    @View.nested
+    class properties(Tab):  # noqa
+        TAB_NAME = 'Properties'
+        instance_type = BootstrapSelect('hardware__instance_type')
+        guest_keypair = BootstrapSelect('hardware__guest_access_key_pair')
+        hardware_monitoring = BootstrapSelect('hardware__monitoring')
+        boot_disk_size = BootstrapSelect('hardware__boot_disk_size')
+        # GCE
+        is_preemtible = VersionPick({
+            Version.lowest(): None,
```

Comment:
I am not sure how that is gonna be handled. It might need some more investigation :)